src/Controller/Component/CsrfComponent.php

```diff
@@ -61,6 +61,19 @@ class CsrfComponent extends Component
         'field' => '_csrfToken',
     ];
 
+    /**
+     * Warn if CsrfComponent is used together with CsrfProtectionMiddleware
+     *
+     * @param array $config The config data.
+     * @return void
+     */
+    public function initialize(array $config)
```

Comment:
we have `startup` callback .  why new one ?

Reply:
I like to warn as possible. 
Also at this place `loadComponent()` is still in callstack so it might be easier to find location of that load. `deprecationWarning()` has `$stackFrame` parameter, but it should `5`, so I wasn't sure if such depth is reliable, so I used `triggerWarning()` but it stayed here.